rust/src/treefile.rs

```diff
@@ -576,10 +576,42 @@ impl TreefileExternals {
 /// array elements.
 fn whitespace_split_packages(pkgs: &[String]) -> Vec<String> {
     pkgs.iter()
-        .flat_map(|pkg| pkg.split_whitespace().map(String::from))
+        .flat_map(|element| split_whitespace_unless_quote(element).map(String::from))
         .collect()
 }
 
+// Helper for whitespace_split_packages().
+// Splits a String by whitespace unless substring is wrapped between single quotes
+// and returns split Strings in an Iterator.
+fn split_whitespace_unless_quote(element: &String) -> std::vec::IntoIter<String> {
```

Comment:
This one should be easy to write unit test(s) for via `#[test]`.

Reply:
Right. I'll look into how unit tests work in Rust.

Reply:
What you have is fine, but I'd say it's more idiomatic to write `impl IntoIterator<Item=String> {` for the return value.  A bit more info here https://doc.rust-lang.org/edition-guide/rust-2018/trait-system/impl-trait-for-returning-complex-types-with-ease.html